### ISSUE TITLE
508-defect-2 [SCREENREADER, COGNITION]: Table labels SHOULD be concise and descriptions SHOULD use aria-describedby - eBenefits View Payments

### ISSUE THREAD

**Issue by micahchiang** [department-of-veterans-affairs/va.gov-team#18518]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I expect to hear concise labels for tables so that I can understand them efficiently.

## Details

As of now, tables are labelled by two `aria-labelledby` ids:
- The heading of the section 👍 
- The description of the section 👎 

Because the description of the section is added to the label, the table's name becomes too verbose for screen reader users.

<img width="300" alt="Screen Shot 2020-12-09 at 9 32 34 AM" src="https://user-images.githubusercontent.com/14154792/101642849-7a875580-3a01-11eb-97ff-0aae9863ca6e.png">

This can easily be fixed by appropriately mapping the description of the section to `aria-describedby` instead of appending it to the label.

Furthermore, the table does not describe to the screen reader user the string "Displaying X - Y of Z" which is valuable information. Sighted users will be able to notice this change every time they move through the pagination or if they were to jump directly to the table itself using a screen reader shortcut. In order to provide this to our screen reader users we should:
- Add an `aria-live="polite"` to this `p`
- Add a unique id to this `p` and use it to describe the table using `aria-describedby`

## Tasks
- [ ] Update View Payments table headers once the responsive table formation react component has been updated.

## Acceptance Criteria
- [ ] Both table's `aria-labelledby` matches the `id` of the section's heading.
- [ ] Both table's `aria-describedby` matches the description below the section's heading and announces what results are being displayed.

## Environment

* Operating System: Mac, Windows
* Browser: Safari, Firefox
* Screenreading device: VoiceOver, NVDA
* Server destination: staging, production

## Steps to Recreate

1. [Log into user 54's account and open the payment history page](https://staging.va.gov/va-payment-history/payments/?postLogin=true).
2. Inspect the table using developer tools.
3. Confirm it has an `aria-labelledby` that points to both the heading and description of the section.

## Solution

There are two tables on this page, so don't forget to update both of them 😄 

### Before
```html
<!-- The paragraph that describes how many items exist does not semantically describe the table (yet) -->
<p class="vads-u-font-size--lg vads-u-font-family--serif">Displaying 55 - 60 of 67</p>

<!-- The table includes its description under aria-labelled which we'll have to change -->
<table 
  aria-labelledby="paymentsRecievedHeader paymentsRecievedContent" 
  class="responsive" 
  role="table">
```

### After
``` diff
<!-- Add a unique id to this paragraph that describes how many items are displaying -->
<!-- I've also added "payments" to this string so it's clear to the user what is being displayed -->
<!-- To ensure this gets announced when the user navigates the pagination, I've also added aria-live-->
+ <p id="unique-id" aria-live="polite" class="vads-u-font-size--lg vads-u-font-family--serif">
+     Displaying 55 - 60 of 67 payments.
+ </p>

<!-- Add the unique id into an aria-describedby and move the descriptive id from aria-labelledby into aria-describedby.-->
<table 
  aria-labelledby="paymentsRecievedHeader" 
+  aria-describedby="unique-id paymentsRecievedContent" 
  class="responsive" 
  role="table">
```

Some notes here:
- I've placed the "Displaying 55-60..." string ahead of the longer table description so it does not get buried in verbose content.
- While testing, I noticed that adding an `id` and `aria-live` to the `p` in question causes it to stop updating. I'm unsure as to the reason why and could use some help debugging there.

## WCAG or Vendor Guidance

[WAI-ARIA 1.1: The aria-labelledby attribute is similar to aria-describedby in that both reference other elements to calculate a text alternative, but a label should be concise, where a description is intended to provide more verbose information.](https://www.w3.org/TR/wai-aria/#aria-labelledby)

### Before
<img width="800" alt="Screen Shot 2020-12-09 at 10 31 21 AM" src="https://user-images.githubusercontent.com/14154792/101650337-d0f89200-3a09-11eb-837b-6ef0d5ac2f04.png">

### After
<img width="786" alt="Screen Shot 2020-12-09 at 10 32 00 AM" src="https://user-images.githubusercontent.com/14154792/101650269-c6d69380-3a09-11eb-86b8-585c6eefa568.png">

Adding "payments" to this string will make the description more clear to all users. @sporeboy As a design consideration, I'd also suggest reducing the sizing of this paragraph so it does not compete for attention on this page as it is descriptive (as opposed to being a heading).

**Comment by jason-gcio** [comment]
We will present this to platform to see if they have any guidance or next steps

**Comment by micahchiang** [comment]
Seems like an exact duplicate of #17149. Closing.

**Comment by jason-gcio** [comment]
@cohnjesse - whenever you get a chance, can you leave a comment as to where this ticket stands with its next steps?  I think we are waiting to hear back from platform but wasn't sure if it was this one or the other one...

**Comment by jason-gcio** [comment]
@joshkimux - we are seeing this as a duplicate to #17149, can you confirm that for us?

**Comment by joshkimux** [comment]
@jason-gcio I think two ended up being made to keep track of where the issue originally stemmed from. So yes, this is a duplicate!
